app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java

```diff
@@ -0,0 +1,23 @@
+package ru.yandex.practicum.contacts.presentation.base;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.DiffUtil;
+
+public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T> {
+    @Override
+    public boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem) {
+        return oldItem.theSameAs(oldItem) == newItem.theSameAs(newItem);
```

Comment:
Здесь всегда будет true, потому что объект `oldItem` всегда равен самому себе, как и `newItem`. Сравнить нужно два объекта `oldItem` и `newItem` друг с другом:
```suggestion
        return oldItem.theSameAs(newItem);
```